src/LogicTasks/Syntax/DecomposeFormula.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module LogicTasks.Syntax.DecomposeFormula where
+
+
+import Control.Monad.Output (
+  LangM,
+  OutputMonad, german, english,
+  GenericOutputMonad (paragraph, indent, translatedCode, refuse, image), translate, localise, translations, ($=<<)
+  )
+
+import LogicTasks.Helpers (extra, example, instruct, keyHeading, basicOpKey, reject)
+import Trees.Types (TreeFormulaAnswer(..))
+import Tasks.DecomposeFormula.Config (DecomposeFormulaInst(..), DecomposeFormulaConfig, checkDecomposeFormulaConfig)
+import Trees.Print (display, transferToPicture)
+import Control.Monad (when, unless)
+import Data.Maybe (isNothing, fromJust)
+import Trees.Helpers (collectLeaves, collectUniqueBinOpsInSynTree, swapKids)
+import Data.Containers.ListUtils (nubOrd)
+import Control.Monad.Cont (MonadIO (liftIO))
+import LogicTasks.Syntax.TreeToFormula (cacheTree)
+
+
+
+
+description :: OutputMonad m => DecomposeFormulaInst -> LangM m
+description DecomposeFormulaInst{..} = do
+
+    example (display tree) $ do
+      english "The following formula is given:"
+      german "Die folgende Formel ist gegeben:"
+
+    instruct $ do
+      english "Create the syntax tree for this formula and swap the two subtrees at the root. "
+      english "Finally, enter the formula for the resulting syntax tree."
+      german "Erstellen Sie den Syntaxbaum zu dieser Formel und tauschen Sie die beiden Teilbäume an der Wurzel. "
+      german "Geben Sie anschließend die Formel für den entstandenen Syntaxbaum ein."
+
+    instruct $ do
+      english "(You are allowed to add arbitrarily many additional pairs of brackets in the formula.)"
+      german "(In der Formel dürfen Sie beliebig viele zusätzliche Klammerpaare hinzufügen.)"
+
+    when addExtraHintsOnAssociativity $ instruct $ do
+      english "Remark: Do not try to use associativity in this task in order to save brackets."
+      german "Hinweis: Sie dürfen bei dieser Aufgabe nicht Klammern durch Verwendung von Assoziativität einsparen."
+
+    keyHeading
+    basicOpKey
```

Comment:
What about `arrowsKey`?

Reply:
Result of https://github.com/fmidue/logic-tasks/pull/122#issuecomment-2037252458